torch/nn/_functions/vision.py

```diff
@@ -9,50 +9,24 @@
 class GridSampler(Function):
 
     @staticmethod
-    def _enforce_cudnn(input):
-        if not cudnn.enabled:
-            raise RuntimeError("GridSampler needs CuDNN for processing CUDA inputs,"
-                               " but CuDNN is not enabled")
-        assert cudnn.is_acceptable(input)
-
-    @staticmethod
     def forward(ctx, input, grid):
         ctx.save_for_backward(input, grid)
         grid_sz = grid.size()
-        if input.is_cuda:
-            GridSampler._enforce_cudnn(input)
-            output = input.new(grid_sz[0], input.size(1), grid_sz[1], grid_sz[2])
-            grid = grid.contiguous()
-            if 0 in input.stride():
-                input = input.contiguous()
-            torch._C._cudnn_grid_sampler_forward(input, grid, output)
-        else:
-            backend = type2backend[type(input)]
-            output = input.new(grid_sz[0], input.size(1), grid_sz[1], grid_sz[2])
-            backend.SpatialGridSamplerBilinear_updateOutput(backend.library_state, input, grid, output)
+        backend = type2backend[type(input)]
```

Comment:
if cudnn is available, still use that. If it's not available, only then do you use the THCUNN implementation